include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -86,13 +85,33 @@ struct is_operational_params
      */
     bdl_input_iterator_params input_bdl_iterator_params{};
     /**
-     * Condition which is used to decide if a layout is `operational` or `non-operational`.
+     * Condition which is used to decide if a layout is operational or non-operational.
      */
     operational_condition op_condition = operational_condition::TOLERATE_KINKS;
 };
 
 namespace detail
 {
+
+/**
+ * Reason why a layout is non-operational.
+ */
+enum class non_operationality_reason : uint8_t
+{
+    /**
+     * Kinks induced the layout to become non-operational.
+     */
+    KINKS,
+    /**
+     * The layout is non-operational due to a logic mismatch.
```

Comment:
```suggestion
     * The layout is non-operational because of a logic mismatch.
```
Not a necessity to reword here, but I like this formulation a bit better :laughing: Up to you. It's your code.